### ISSUE TITLE
TilingSprite is showing a gap

### ISSUE THREAD

**Issue by ErikSom** [pixijs/pixijs#7625]
### Expected Behavior
Seamless tiling of textures without gaps
![image](https://user-images.githubusercontent.com/1382022/126512965-05e42dd7-ee20-472b-b8d7-c3fb1ba164bb.png)

### Current Behavior
Showing gaps between the tiles
![image](https://user-images.githubusercontent.com/1382022/126511859-53f85611-f86b-4e7a-adf1-dea2f6db53aa.png)

### Steps to Reproduce
Run this: https://replit.com/@ErikSombroek/Pixi-TilingSprite-bug

### Potential Solution
Is inside the repro

### Environment
- **`pixi.js` version**: latest
- **Browser & Version**: Chrome 91.0.4472.114
- **OS & Version**: _e.g. Mac Catalina 10.15.7 
- **Running Example**: https://replit.com/@ErikSombroek/Pixi-TilingSprite-bug

**Comment by ivanpopelyshev** [comment]
mipmap + atlas again. Need to make new TilingSprite shader that doesnt care about mip levels.

**Comment by ShukantPal** [comment]
TLDR; The best way to solve this issue on your end @ErikSom is to disable mipmapping.

```
import { MIPMAP_MODES } from '@pixi/constants';

baseTexture.mipmap = MIPMAP_MODES.OFF;
```

I spent a few hours, with the help of @ivanpopelyshev's wisdom, working on a solution for this. It turns out it's not 100% solvable without changing how users are atlasing (which I'll get to later).

The root cause of the issue is that at the "border" of the texture, the change in UV coordinates (e.g. du / dx) is very large. The GPU interprets this as very significant texture _minification_, causing it to sample to smallest LOD. When this happens, the color is basically the average of the whole atlas.

In the following screen capture, I mark the areas with alpha < 0 sampling as red:

https://user-images.githubusercontent.com/22450567/133008806-7147724c-60c9-4115-8ce6-125f61a9eb93.mov

On devices with [EXT_shader_texture_lod](https://www.khronos.org/registry/webgl/extensions/EXT_shader_texture_lod/), we can solve this by explicitly requesting LOD 0 using `texture2DLod`. On WebGL 2, this can be fixed by sampling using the `texture` function with a very negative LOD bias. Using the 0-LOD is desired only when sampling the _edge_ of the texture (because that's where the wrapping occurs, causing large du/dx, dv/dy values)

This is the WebGL 2 shader running:

https://user-images.githubusercontent.com/22450567/133010188-000deb5e-fe37-4368-90ba-8e98b00a42f6.mov

**Caveat**: Looks like macOS Safari neither supports the WebGL 1 extension nor WebGL 2 - so you'll have to turn mipmapping forcefully on that platform to fix this.
